packages/identity/src/test/index.test.ts

```diff
@@ -22,23 +21,49 @@ describe('Message signature verification', () => {
 
   it('returns true if public key and message signature are correct', async () => {
     const message = 'hello'
-    const dmPublicKey = Buffer.from('0bfb475810c0e26c9fab590d47c3d60ec533bb3c451596acc3cd4f21602e9ad9', 'hex')
-    const dmPublicKeyString = dmPublicKey.toString()
-    const dmPublicKeyArrayBuffer = stringToArrayBuffer(dmPublicKeyString)
+
     const rootCert = await createTestRootCA()
     const userCsr = await createTestUserCsr()
     const userCert = await createTestUserCert(rootCert, userCsr)
 
     const data = {
       message: message,
       userPubKey: await extractPubKey(userCert.userCertString, crypto),
-      signature: await sign(message, userCsr.pkcs10.privateKey),
-      dmPublicKey: dmPublicKeyArrayBuffer
+      signature: await sign(message, userCsr.pkcs10.privateKey)
     }
 
-    const result = await verifySignature(data.userPubKey, data.signature, data.message)
+    const result = await verifySignature(
+      data.signature,
+      data.message,
+      data.userPubKey
+    )
+
     expect(result).toBe(true)
   })
+
+  it("returns false if public key and message signature doesn't match", async () => {
+    const message = 'spoofed'
+
+    const rootCert = await createTestRootCA()
+    const userCsr = await createTestUserCsr()
+
+    const spoofedUserCsr = await createTestUserCsr()
+    const spoofedUserCert = await createTestUserCert(rootCert, spoofedUserCsr)
+
+    const data = {
+      message: message,
+      userPubKey: await extractPubKey(spoofedUserCert.userCertString, crypto),
+      signature: await sign(message, userCsr.pkcs10.privateKey)
+    }
+
+    const result = await verifySignature(
+      data.signature,
+      data.message,
+      data.userPubKey
+    )
+
+    expect(result).not.toBe(true)
```

Comment:
`.not.toBe(true)` means that if `result === 'Różowy słoń'` it will still pass the test

I suggest 
```
toBe(false)
```

Reply:
I keep forgetting, in typescript boolean variable can be boolean. Or string. Or object...